mongo/collection.go

```diff
@@ -319,7 +319,9 @@ func (coll *Collection) insert(ctx context.Context, documents []interface{},
 //
 // The document parameter must be the document to be inserted. It cannot be nil. If the document does not have an _id
 // field when transformed into BSON, one will be added automatically to the marshalled document. The original document
-// will not be modified. The _id can be retrieved from the InsertedID field of the returned InsertOneResult.
+// will not be modified. The _id can be retrieved from the InsertedID field of the returned InsertOneResult. The document
+// parameter should NOT contain duplicate key names, as that can result in undefined behavior from the server.
```

Comment:
Would this apply to any CRUD function that results in a document getting inserted or replaced? (`FindOneAndReplace`, `ReplaceOne`, `Replace`). As well as the insert models for bulk write.

If that is right, then I think we'd want to document them all, or just document `bson.D`. My vote goes for just `bson.D`.

Reply:
Discussed offline: it seems like overkill to document every CRUD function that results in a document getting inserted or replaced. Instead, we've opted to only warn against duplicate keys on `bson.D` and in `bson/doc.go`. Let me know what you think...

Reply:
I prefer this for consistency.